examples/ownable-two-step/tests/ownable_two_step.rs

```diff
@@ -0,0 +1,237 @@
+#![cfg(feature = "e2e")]
+
+use abi::{
+    Ownable2Step,
+    Ownable2Step::{OwnershipTransferStarted, OwnershipTransferred},
+};
+use alloy::{primitives::Address, sol};
+use e2e::{receipt, send, Account, EventExt, ReceiptExt, Revert};
+use eyre::Result;
+
+use crate::Ownable2StepExample::constructorCall;
+
+mod abi;
+
+sol!("src/constructor.sol");
+
+fn ctr(owner: Address) -> constructorCall {
+    constructorCall { initialOwner: owner }
+}
+
+// ============================================================================
+// Integration Tests: Ownable2Step
+// ============================================================================
+
+#[e2e::test]
+async fn constructs(alice: Account) -> Result<()> {
+    let alice_addr = alice.address();
+    let receipt =
+        alice.as_deployer().with_constructor(ctr(alice_addr)).deploy().await?;
+    let contract = Ownable2Step::new(receipt.address()?, &alice.wallet);
+
+    assert!(receipt.emits(OwnershipTransferred {
+        previousOwner: Address::ZERO,
+        newOwner: alice_addr,
+    }));
+
+    let Ownable2Step::ownerReturn { owner } = contract.owner().call().await?;
+    assert_eq!(owner, alice_addr);
+    let Ownable2Step::pendingOwnerReturn { pendingOwner } =
+        contract.pendingOwner().call().await?;
+    assert_eq!(pendingOwner, Address::ZERO);
+
+    Ok(())
+}
+
+#[e2e::test]
+async fn construct_reverts_when_owner_is_zero_address(
+    alice: Account,
+) -> Result<()> {
+    let err = alice
+        .as_deployer()
+        .with_constructor(ctr(Address::ZERO))
+        .deploy()
+        .await
+        .expect_err("should not deploy due to `OwnableInvalidOwner`");
+
+    assert!(err.reverted_with(Ownable2Step::OwnableInvalidOwner {
+        owner: Address::ZERO
+    }));
+    Ok(())
+}
+
+#[e2e::test]
+async fn transfer_ownership_initiates_transfer(
```

Comment:
Missing tests for cancelling transfer ownership and overwriting previous transfer ownership

Reply:
Solved by #399